datamodels/2.x/itop-portal-base/portal/src/Controller/ObjectController.php

```diff
@@ -1263,6 +1266,12 @@ public function GetInformationAsJsonAction(Request $oRequest)
 		// Retrieving objects
 		while ($oObject = $oSet->Fetch())
 		{
+            $sObjectId = $oObject->Get('id');
+            if (!$oSecurityHelper->IsActionAllowed(UR_ACTION_READ, $sObjectClass, $sObjectId))
+            {
+                IssueLog::Warning(__METHOD__.' at line '.__LINE__.' : User #'.UserRights::GetUserId().' not allowed to read '.$sObjectClass.'::'.$sObjectId.' object.');
```

Comment:
Reduce var scope and use specific method.
```suggestion
            if (!$oSecurityHelper->IsActionAllowed(UR_ACTION_READ, $sObjectClass, $sObjectId))
            {
                $iObjectId = $oObject->GetKey();
                IssueLog::Warning(__METHOD__.' at line '.__LINE__.' : User #'.UserRights::GetUserId().' not allowed to read '.$sObjectClass.'::'.$iObjectId.' object.');
```

Reply:
`$iObjectId` is used in the `$oSecurityHelper->IsActionAllowed()` condition, so you can't put it inside the block

Reply:
Oh, I missed that. Well it could still be changed to use the `GetKey()` method 😉

Reply:
Yep that would be correct too but GetKey() returns an int, while IsActionAllowed expect (given parameter name) a string

Reply:
Moreover, I prefer storing a value in a variable instead of calling a method twice (even if `GetKey()` isn't very costly, it's always cheaper to access a variable)

Reply:
Yes, keep using the variable is fine, never argued that 😉

Reply:
I know it was just the variable naming, but keep in mind that unfortunately there is no consistancy in the app for the expected type of an object key. Someplaces it's an int, others it's a string.